README.md

```diff
@@ -1,79 +1,132 @@
+
+Varies by usage, but is based on [`key-name`](#key-name) specified in environment variables and the `input-file` specified.
 
 ## Usage Examples
 
 ```yml
-# TODO: Fill in the correct usage
+...
+input:
+  environment:
+    description: The environment to deploy to.
+    required: true
+
 jobs:
-  job1:
+  setup:
     runs-on: ubuntu-20.04
-    steps:
-      - uses: actions/checkout@v2
+    output:
+      env-scope: ${{ steps.env-scope.output.ENVIRONMENT }}
 
-      - name: Add Step Here
-        uses: im-open/this-repo@v1.0.0
+    steps:
+      - name: Set environment scope
+        id: env-scope
+        uses: im-open/set-environment-variables-by-scope@v1.0.0
+        with:
+          scope: ${{ workflow.inputs.environment }}
+          create-output-variables: true
+        env:
+          ENVIRONMENT@dev d development: dev
+          ENVIRONMENT@qa a: qa
+          ENVIRONMENT@stage s stg: stage
+          ENVIRONMENT@prod p production: prod
+
+      - run: echo "The current environment is ${{ env.ENVIRONMENT }}.
+
+    build:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Build Workflow Environment Variables
+        uses: im-open/set-environment-variables-by-scope@v1.0.0
+        with:
+          scope: ${{ needs.setup.outputs.env-scope }}
+          input-file: ./env-vars.yml
+        env:
+          keyName1@dev qa stage prod: 'key value 1'
+          keyName2@dev qa: 'key value 2 lower'
+          keyName2@stage prod: 'key value 2 upper'
+
+      - name: Display Env Var
+        uses: actions/github-script@v5
         with:
-          input-1: 'abc'
-          input-2: '123
+          script: |
+            console.log("Env", ${{ needs.setup.outputs.env-scope }})
+            console.log("keyName1", ${{ env.keyName1 }})
+            console.log("keyName2", ${{ env.keyName2 }})
+            console.log("inFileName1", ${{ env.inFileName1 }})
+            console.log("inFileName2", ${{ env.inFileName2 }})
+...
+```
+
+### Usage Instructions
+
+The format of the `env` variable and the `input-file` contents are the same and follows this format:
+
+`[key-name]@[scope-array]: [key-value]`
+
+#### `key-name`
+
+The resulting environment variable name. It does accept a wide variety of name convention formats including spaces, dashes, periods, and underscores. While it's been tested and found to be fairly flexible, it is possible to make environment variable names that aren't usable by one or more of the target action runner operating systems.
+
+**_Key names intended for use as output variables for in workflow references can only contain letters, numbers, dashes and underscores._** Any other punctuation or space characters cannot processed and will cause errors when attempting to reference them later in a workflow.
+
+#### `scope-array`
+
+A space delimited array of scope strings that the environment variable could be valid for.  The filter for which keys/values will be select from is the `scope` action input.
```

Comment:
The second sentence reads a bit strangely to me. This was my stab at trying to make it better, but feel free to do something else.
```suggestion
A space delimited array of scope strings that the environment variable could be valid for.  The filter by which keys/values will be selected is the `scope` action input.
```